address_book_query.go

```diff
@@ -0,0 +1,182 @@
+package hedera
+
+import (
+	"context"
+	"github.com/hashgraph/hedera-protobufs-go/services"
+	"io"
+	"math"
+	"time"
+
+	"github.com/hashgraph/hedera-protobufs-go/mirror"
+	"google.golang.org/grpc/status"
+)
+
+//var rstStream = regexp.MustCompile("(?i)\\brst[^0-9a-zA-Z]stream\\b")
+
+type AddressBookQuery struct {
+	errorHandler      func(stat status.Status)
+	completionHandler func()
+	retryHandler      func(err error) bool
+	attempt           uint64
+	maxAttempts       uint64
+	fileID            *FileID
+	limit             int32
+}
+
+func NewAddressBookQuery() *AddressBookQuery {
+	return &AddressBookQuery{
+		fileID: nil,
+		limit:  0,
+	}
+}
+
+func (query *AddressBookQuery) SetFileID(id FileID) *AddressBookQuery {
+	query.fileID = &id
+	return query
+}
+
+func (query *AddressBookQuery) GetFileID() FileID {
+	if query.fileID == nil {
+		return FileID{}
+	}
+
+	return *query.fileID
+}
+
+func (query *AddressBookQuery) SetLimit(limit int32) *AddressBookQuery {
+	query.limit = limit
+	return query
+}
+
+func (query *AddressBookQuery) GetLimit() int32 {
+	return query.limit
+}
+
+func (query *AddressBookQuery) SetMaxAttempts(maxAttempts uint64) *AddressBookQuery {
+	query.maxAttempts = maxAttempts
+	return query
+}
+
+func (query *AddressBookQuery) GetMaxAttempts() uint64 {
+	return query.maxAttempts
+}
+
+func (query *AddressBookQuery) SetErrorHandler(errorHandler func(stat status.Status)) *AddressBookQuery {
+	query.errorHandler = errorHandler
+	return query
+}
+
+func (query *AddressBookQuery) SetCompletionHandler(completionHandler func()) *AddressBookQuery {
+	query.completionHandler = completionHandler
+	return query
+}
+
+func (query *AddressBookQuery) SetRetryHandler(retryHandler func(err error) bool) *AddressBookQuery {
+	query.retryHandler = retryHandler
+	return query
+}
+
+func (query *AddressBookQuery) _ValidateNetworkOnIDs(client *Client) error {
+	if client == nil || !client.autoValidateChecksums {
+		return nil
+	}
+
+	if query.fileID != nil {
+		if err := query.fileID.ValidateChecksum(client); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (query *AddressBookQuery) _Build() *mirror.AddressBookQuery {
+	body := &mirror.AddressBookQuery{
+		Limit: query.limit,
+	}
+	if query.fileID != nil {
+		body.FileId = query.fileID._ToProtobuf()
+	}
+
+	return body
+}
+
+func (query *AddressBookQuery) Subscribe(client *Client) (NodeAddressBook, error) {
```

Comment:
I think this query should be as similar to our regular queries as we can get; meaning we should have an `Execute` method.